client/src/components/Attachment/AttachmentsDetailView.tsx

```diff
@@ -0,0 +1,84 @@
+import AppContext from "components/AppContext"
+import UploadAttachment from "components/Attachment/UploadAttachment"
+import React, { useContext, useState } from "react"
+import { Button } from "react-bootstrap"
+import Settings from "settings"
+import AttachmentCard from "./AttachmentCard"
+
+interface AttachmentsListProps {
+  attachments: any[]
+}
+
+const AttachmentsList = ({ attachments }: AttachmentsListProps) => {
+  if (attachments.length === 0) {
+    return <em>No attachments found</em>
+  }
+  return (
+    <div className="attachment-card-list">
+      {attachments.map(attachment => (
+        <AttachmentCard key={attachment.uuid} attachment={attachment} />
+      ))}
+    </div>
+  )
+}
+
+interface AttachmentsDetailViewProps {
+  attachments: any[]
+  updateAttachments: (attachments: any[]) => void
+  relatedObjectType: string
+  relatedObjectUuid: string
+  allowEdit?: boolean
+}
+
+const AttachmentsDetailView = ({
+  attachments,
+  updateAttachments,
+  relatedObjectType,
+  relatedObjectUuid,
+  allowEdit = false
+}: AttachmentsDetailViewProps) => {
+  const [editAttachments, setEditAttachments] = useState(false)
+
+  const { currentUser } = useContext(AppContext)
+  const isAdmin = currentUser && currentUser.isAdmin()
```

Comment:
`currentUser && currentUser.isAdmin()` is better written as `currentUser?.isAdmin()`.